tools/git.py

```diff
@@ -38,6 +41,11 @@ def run(self):
         args += [repository_dir]
         logging.info("running command: %s" % " ".join(args))
         sp.run(args, check=True)
+
+        if self.clone_submodules:
```

Comment:
IMHO it would be easier to directly call:

`git clone --recurse-submodules`

Reply:
although this might run into the same issue as mentioned by Willi

Reply:
I did some research and the same flag also exists for checkout

> ```git checkout --recurse-submodules```
Using --recurse-submodules will update the content of all initialized submodules according to the commit recorded in the superproject.
If local modifications in a submodule would be overwritten the checkout will fail unless -f is used.

so you could change the PR in a way that if the flag `clone_submodules` is set, then `--recurse-submodules` is added both to clone and checkout.